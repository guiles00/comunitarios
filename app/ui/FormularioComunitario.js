import React, { useEffect } from "react";
import { Link } from "react-router-dom";
import axios from "axios";
import { startUpdateComunitarios } from "../actions/comunitariosActions";

import { useSelector, useDispatch} from "react-redux";


/** OJO QUE GUARDA EN El reducer guarda en Sting y no Int */
const FormularioComunitario = function FormularioComunitario(props) {

  const [comunitario, setComunitario] = React.useState({_id:null,nombre:"",doppler:0,bidi:0,doble:0,consultorio:0});
  const [error, setError] = React.useState("");
 
  const comunitarios = useSelector(state => state.comunitarios.listaComunitarios);
  const isLoading = useSelector(state => state.common.isLoading); 
  const dispatch = useDispatch();

  const com = comunitarios.filter((c)=>c._id === props.match.params.id);

  useEffect(() => {

    if( props.match.params.id !== "new") {
      const { _id, nombre } = com[0];
      const { doppler, doble, bidi, consultorio } = com[0].estudios[com[0].estudios.length -1];

      setComunitario({_id, nombre,doppler,bidi,doble,consultorio})
    }
      
  },[]);

  const handleSubmit = function(e) {
    e.preventDefault();
    //setIsloading(true);

    if(comunitario._id === null){

      axios({
        method: "post",
        url: "/api/comunitarios",
        data: { nombre: comunitario.nombre, doppler: comunitario.doppler, bidi: comunitario.bidi
          , doble: comunitario.doble, consultorio: comunitario.consultorio}
        }).then((res)=>{
          //setIsloading(false);     
          props.history.push("/comunitarios");
          
        }).catch((e)=>{
          console.log(e)
          //setIsloading(false);
          setError("ocurrio un error")
        });

    } else {
      console.log("actualiza esto");
      console.log(comunitario);
      dispatch(startUpdateComunitarios(comunitario))

      // axios.put(`/api/comunitarios/${comunitario._id}`, comunitario)
      //   .then(() => {
      //     setIsloading(false);
      //   }).catch(e => {
      //     console.log(e);
      //     setError("ocurrio un error")
      //   });
    }
  };

  const handleInputChange = function(e) {
    const { name, value } = e.target;

    setComunitario({ ...comunitario, [name]: value });
  };

  const isDisabled = ()=>{
    let deshabilitar = false;

    if(comunitario.nombre === "") deshabilitar = true; 

    return deshabilitar;   
  }

  return (
    <div className="container-fluid pull-down" id="comunitario-form">
      <div className="row justify-content-md-center">
        <div className="col-lg-8 col-sm-12">
          <form>
            <fieldset>
              <div className="card">
                <div className="card-body justify-content-md-center center_div">
                  <h4 className="card-title text-center">Agregar Comunitario</h4>
                  <hr/>
                  {error&&
                    <div className="alert alert-danger" role="alert">
                      {error}
                    </div>}
                    {isLoading &&
                    <div className="spinner-border" role="status">
                        <span className="sr-only">Loading...</span>
                    </div>
                    }
                  <div className="row">
                    <div className="col-sm-12 col-lg-12">
                      <label>Nombre:</label>
                      <input type="text" className="form-control" id="nombre" name="nombre" value={comunitario.nombre} onChange={handleInputChange}></input>
                    </div>
                  </div>
                  <div className="row">
                    <div className="col-sm-12 col-lg-12">
                      <label>Doppler:</label>
                      <input type="text" className="form-control" name="doppler" value={comunitario.doppler} onChange={handleInputChange}></input>
                    </div>
                  </div>
                  <div className="row">
                    <div className="col-sm-12 col-lg-12">
                      <label>BIDI:</label>
                      <input type="text" className="form-control" name="bidi" value={comunitario.bidi} onChange={handleInputChange}></input>
                    </div>
                  </div>
                  <div className="row">
                    <div className="col-sm-12 col-lg-12">
                      <label>Doble:</label>
                      <input type="text" className="form-control" name="doble" value={comunitario.doble} onChange={handleInputChange}></input>
                    </div>
                  </div>
                  <div className="row">
                    <div className="col-sm-12 col-lg-12">
                      <label>Consultorio:</label>
                      <input type="text" className="form-control" name="consultorio" value={comunitario.consultorio} onChange={handleInputChange}></input>
                    </div>
                  </div>
                </div>
                
                <div className="card-footer text-muted">
                  <div className="form-group">
                    <button className="btn btn-icon btn-primary" onClick={handleSubmit} disabled={isDisabled()}>
                      Guardar
                    </button> &nbsp;
                    <Link to={"/comunitarios"} className="btn btn-dark">Volver</Link> &nbsp;&nbsp;
                    <button type="submit" className="btn btn-icon btn-danger">
                      Eliminar
                    </button> &nbsp;
                  </div>
                </div>
              </div>

            </fieldset>
          </form>
        </div>
      </div>
    </div>
   
  );
};

export default FormularioComunitario;