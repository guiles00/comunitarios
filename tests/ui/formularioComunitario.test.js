import Enzyme from 'enzyme';
import { shallow } from "enzyme";
import React  from "react";

import FormularioComunitario from "../../app/ui/FormularioComunitario";


describe("Checking FormularioComunitario",()=>{
 
    // test("Should mount ok",()=>{
    //   const wrapper = shallow(<FormularioComunitario />);
    // });
    
    // test("Should mount ok with data", ()=>{ //esto no funciona
    //     const wrapper = shallow(<FormularioComunitario comunitario={{_id:null,nombre:"Guiles",doppler:100,bidi:100,doble:100,consultorio:0}} />);
    // });

    // test("Should update nombre",()=>{
    // const wrapper = shallow(<FormularioComunitario />);
    // });
  test("",()=>{})
})

// describe('<FormularioComunitario />', () => {
  
//     it('calls componentDidMount', () => {
//     sinon.spy(FormularioComunitario);
//     const wrapper = mount(<FormularioComunitario />);
//     //expect(Foo.prototype.componentDidMount).to.have.property('callCount', 1);
//   });

// });
// // test("component should mount ok", ()=>{
// //     //const wrapper = shallow(<FormularioComunitario />);
// //    // const wrapper = mount(<FormularioComunitario />);

// // });


test("Expect listado de comunitarios",() =>{
    expect(1).toBe(1);
});


describe("Formulario Comunitario",()=>{
  let mockSetComunitario = jest.fn();
  let wrapper;

  const setup = ()=>{
    return shallow(<FormularioComunitario />)
  }

  beforeEach(()=>{
    mockSetComunitario.mockClear();
    React.useState = jest.fn( ()=>[{},mockSetComunitario] );
    wrapper = setup();
  }) 

  test("renders ok",()=>{
    const wrapper = setup();
    expect(wrapper.find("#comunitario-form").length).toBe(1);
  });

  test("Should update comunitario state",()=>{
   //cambiar para que encuentre por nombre
    //const  inputName = wrapper.find("#nombre");
    const inputName = wrapper.find({ name: "nombre" });
    
    const mockEvent = {target: { value:"Comunitario", name:"nombre" } };
    inputName.simulate("change",mockEvent);
    expect(mockSetComunitario).toHaveBeenCalled()
  
    expect(mockSetComunitario).toHaveBeenCalledWith({nombre:"Comunitario"})
  
  });


})