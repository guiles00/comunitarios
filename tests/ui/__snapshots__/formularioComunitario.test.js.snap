// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking FormularioComunitario Should mount ok 1`] = `
<div
  className="container-fluid pull-down "
>
  <div
    className="row justify-content-md-center"
  >
    <div
      className="col-md-12"
    >
      <form
        onSubmit={[Function]}
      >
        <fieldset>
          <div
            className="card"
          >
            <div
              className="card-body justify-content-md-center center_div"
            >
              <h4
                className="card-title text-center"
              >
                Comunitario
              </h4>
              <hr />
              <div
                className="row"
              >
                <div
                  className="col-6"
                >
                  <label>
                    Nombre:
                  </label>
                  <input
                    className="form-control"
                    name="nombre"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-6"
                >
                  <label>
                    Doppler:
                  </label>
                  <input
                    className="form-control"
                    name="doppler"
                    onChange={[Function]}
                    type="text"
                    value={0}
                  />
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-6"
                >
                  <label>
                    BIDI:
                  </label>
                  <input
                    className="form-control"
                    name="bidi"
                    onChange={[Function]}
                    type="text"
                    value={0}
                  />
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-6"
                >
                  <label>
                    Doble:
                  </label>
                  <input
                    className="form-control"
                    name="doble"
                    onChange={[Function]}
                    type="text"
                    value={0}
                  />
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-6"
                >
                  <label>
                    Consultorio:
                  </label>
                  <input
                    className="form-control"
                    name="consultorio"
                    onChange={[Function]}
                    type="text"
                    value={0}
                  />
                </div>
              </div>
            </div>
            <div
              className="card-footer text-muted"
            >
              <div
                className="form-group"
              >
                <Link
                  className="btn btn-dark"
                  to="/comunitarios"
                >
                  Volver
                </Link>
                   
                <button
                  className="btn btn-icon btn-primary"
                  type="submit"
                >
                  Guardar
                </button>
                  
              </div>
            </div>
          </div>
        </fieldset>
      </form>
    </div>
  </div>
</div>
`;

exports[`Checking FormularioComunitario Should mount ok with data 1`] = `
<div
  className="container-fluid pull-down "
>
  <div
    className="row justify-content-md-center"
  >
    <div
      className="col-md-12"
    >
      <form
        onSubmit={[Function]}
      >
        <fieldset>
          <div
            className="card"
          >
            <div
              className="card-body justify-content-md-center center_div"
            >
              <h4
                className="card-title text-center"
              >
                Comunitario
              </h4>
              <hr />
              <div
                className="row"
              >
                <div
                  className="col-6"
                >
                  <label>
                    Nombre:
                  </label>
                  <input
                    className="form-control"
                    name="nombre"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-6"
                >
                  <label>
                    Doppler:
                  </label>
                  <input
                    className="form-control"
                    name="doppler"
                    onChange={[Function]}
                    type="text"
                    value={0}
                  />
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-6"
                >
                  <label>
                    BIDI:
                  </label>
                  <input
                    className="form-control"
                    name="bidi"
                    onChange={[Function]}
                    type="text"
                    value={0}
                  />
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-6"
                >
                  <label>
                    Doble:
                  </label>
                  <input
                    className="form-control"
                    name="doble"
                    onChange={[Function]}
                    type="text"
                    value={0}
                  />
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-6"
                >
                  <label>
                    Consultorio:
                  </label>
                  <input
                    className="form-control"
                    name="consultorio"
                    onChange={[Function]}
                    type="text"
                    value={0}
                  />
                </div>
              </div>
            </div>
            <div
              className="card-footer text-muted"
            >
              <div
                className="form-group"
              >
                <Link
                  className="btn btn-dark"
                  to="/comunitarios"
                >
                  Volver
                </Link>
                   
                <button
                  className="btn btn-icon btn-primary"
                  type="submit"
                >
                  Guardar
                </button>
                  
              </div>
            </div>
          </div>
        </fieldset>
      </form>
    </div>
  </div>
</div>
`;
